### PR TITLE
feat: Add the ServiceLoadbalancer Addon, with MetalLB as first provider

### PR DESCRIPTION
**Pull request by dlipovetsky**
**What problem does this PR solve?**:
Adds the `ServiceLoadbalancer` addon, and support for MetalLB as the first provider.

### TODO:
- [x] API type
- [x] Generic handler
- [x] MetalLB handler
- [x] Deploy MetalLB Helm chart 
- [x] Default configuration for MetalLB Helm chart
- [ ] Document the ServiceLoadbalancer Addon.

**Which issue(s) this PR fixes**:
Fixes https://jira.nutanix.com/browse/D2IQ-100444

**How Has This Been Tested?**:

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by jimmidyson** [comment]
When you find time, please don't miss out the docs 😉

File: api/v1alpha1/addon_types.go [hunk 1]
```diff
@@ -69,6 +71,9 @@ type Addons struct {
 
 	// +optional
 	CSIProviders *CSI `json:"csi,omitempty"`
+
+	// +optional
+	ServiceLoadbalancer *ServiceLoadbalancer `json:"serviceLoadbalancer,omitempty"`
```
Comment (dkoshkin) [inline api/v1alpha1/addon_types.go:76]: nit: Kubernetes uses "LoadBalancer" or "load balancing"
```suggestion
	ServiceLoadBalancer *ServiceLoadBalancer `json:"serviceLoadBalancer,omitempty"`
```

**Review by dkoshkin — APPROVED** [review]
Looks great, just the one comment about `Loadbalancer` vs `LoadBalancer` in the API, and other vars and output strings.

File: api/v1alpha1/addon_types.go (hunk 1 above)
Comment (jimmidyson) [reply api/v1alpha1/addon_types.go]: Fixed all these up in 410c70174b0968e7a5d77531b6a37d6c5fa0816d

**Review by jimmidyson — APPROVED** [review]

**Review by yannickstruyf3 — APPROVED** [review]

**Pull request merged**